huaweicloud/services/css/resource_huaweicloud_css_snapshot_restore.go

```diff
@@ -35,83 +48,78 @@ func ResourceCssRestore() *schema.Resource {
 			},
+			"enable_force_new": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"true", "false"}, false),
+				Description:  utils.SchemaDesc("", utils.SchemaDescInput{Internal: true}),
+			},
 		},
 	}
 }
 
-func resourceCssRestoreCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func resourceSnapshotRestoreCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	cfg := meta.(*config.Config)
 	region := cfg.GetRegion(d)
-	cssV1Client, err := cfg.CssV1Client(region)
+	client, err := cfg.CssV1Client(region)
 	if err != nil {
 		return diag.Errorf("error creating CSS V1 client: %s", err)
 	}
-	hcCssV1Client, err := cfg.HcCssV1Client(region)
-	if err != nil {
-		return diag.Errorf("error creating CSS V1 client: %s", err)
-	}
-
-	restoreCssCreateHttpUrl := "v1.0/{project_id}/clusters/{cluster_id}/index_snapshot/{snapshot_id}/restore"
 
-	targetClusterID := d.Get("target_cluster_id").(string)
-	backupId := d.Get("snapshot_id").(string)
+	targetClusterId := d.Get("target_cluster_id").(string)
+	snapshotId := d.Get("snapshot_id").(string)
 	clusterId := d.Get("source_cluster_id").(string)
-	restoreCssCreatePath := cssV1Client.Endpoint + restoreCssCreateHttpUrl
-	restoreCssCreatePath = strings.ReplaceAll(restoreCssCreatePath, "{project_id}", cssV1Client.ProjectID)
-	restoreCssCreatePath = strings.ReplaceAll(restoreCssCreatePath, "{cluster_id}", clusterId)
-	restoreCssCreatePath = strings.ReplaceAll(restoreCssCreatePath, "{snapshot_id}", backupId)
+	restoreSnapshotHttpUrl := "v1.0/{project_id}/clusters/{cluster_id}/index_snapshot/{snapshot_id}/restore"
+	restoreSnapshotPath := client.Endpoint + restoreSnapshotHttpUrl
+	restoreSnapshotPath = strings.ReplaceAll(restoreSnapshotPath, "{project_id}", client.ProjectID)
+	restoreSnapshotPath = strings.ReplaceAll(restoreSnapshotPath, "{cluster_id}", clusterId)
+	restoreSnapshotPath = strings.ReplaceAll(restoreSnapshotPath, "{snapshot_id}", snapshotId)
 
-	restoreCssCreateOpt := golangsdk.RequestOpts{KeepResponseBody: true}
-	restoreCssCreateOpt.JSONBody = utils.RemoveNil(buildCreateRestoreBodyParams(d))
+	restoreSnapshotOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+	restoreSnapshotOpt.JSONBody = utils.RemoveNil(buildCreateSnapshotRestoreBodyParams(d))
 
-	_, err = cssV1Client.Request("POST", restoreCssCreatePath, &restoreCssCreateOpt)
+	_, err = client.Request("POST", restoreSnapshotPath, &restoreSnapshotOpt)
 	if err != nil {
 		return diag.Errorf("error restore CSS cluster snapshot extend, cluster_id: %s, error: %s", d.Id(), err)
 	}
 
-	err = checkClusterOperationCompleted(ctx, hcCssV1Client, targetClusterID, d.Timeout(schema.TimeoutCreate))
+	err = checkClusterOperationResult(ctx, client, targetClusterId, d.Timeout(schema.TimeoutCreate))
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	d.SetId(backupId)
+	d.SetId(snapshotId)
```

Comment:
a snapshot_id can be restored for more than 1 cluster, so if a snapshot_id is retored for more than 1 cluster, then the resource ID will not be unique